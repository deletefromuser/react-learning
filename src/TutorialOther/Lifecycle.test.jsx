import renderer, { act } from 'react-test-renderer';
import Lifecycle from './Lifecycle';

it('result', () => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date('2019-01-01T00:00:00.000Z'));

    let component = renderer.create(
        <Lifecycle></Lifecycle>
    );
    let tree = component.toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();

    // expect(component.state('date')).toBe(0);  // Success!
    // jest.advanceTimersByTime(3000);
    jest.setSystemTime(new Date('2019-01-01T00:03:00.000Z'));
    // component.update();
    component = renderer.create(
        <Lifecycle></Lifecycle>
    );
    // expect(component.state('date')).toBe(3);  // Success!
    expect(component.toJSON()).toMatchSnapshot();

    // act(() => {
    //     jest.setSystemTime(new Date('2019-01-01T00:05:00.000Z'));
    //     component.update(<Lifecycle></Lifecycle>);
    // })

    let root;
    act(() => {
        root = renderer.create(
            <Lifecycle></Lifecycle>
        );
    });

    expect(root.toJSON()).toMatchSnapshot();


    jest.useFakeTimers();
    jest.setSystemTime(new Date('2019-01-01T00:05:00.000Z'));
    // renderer.update(component);
    // renderer.
    component.update(<Lifecycle></Lifecycle>);

    act(() => {
        root.update(<Lifecycle></Lifecycle>);
    })


    expect(root.toJSON()).toMatchSnapshot();

});

it('test use act', () => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date('2019-01-01T00:05:00.000Z'));
    // render the component
    let root;
    act(() => {
        root = renderer.create(<Lifecycle />)
    });

    // make assertions on root 
    expect(root.toJSON()).toMatchSnapshot();

    // update with some different props
    act(() => {
        jest.advanceTimersByTime(5000);
        root.update(<Lifecycle />);
    })

    // make assertions on root 
    expect(root.toJSON()).toMatchSnapshot();


});