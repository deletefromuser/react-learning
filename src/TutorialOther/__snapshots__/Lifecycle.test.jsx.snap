// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`result 1`] = `
<div>
  <h1>
    Hello, world!
  </h1>
  <h2>
    It is 
    8:00:00 AM
    .
  </h2>
</div>
`;

exports[`result 2`] = `
<div>
  <h1>
    Hello, world!
  </h1>
  <h2>
    It is 
    8:03:00 AM
    .
  </h2>
</div>
`;

exports[`result 3`] = `
<div>
  <h1>
    Hello, world!
  </h1>
  <h2>
    It is 
    8:03:00 AM
    .
  </h2>
</div>
`;

exports[`result 4`] = `
<div>
  <h1>
    Hello, world!
  </h1>
  <h2>
    It is 
    8:03:00 AM
    .
  </h2>
</div>
`;

exports[`test use act 1`] = `
<div>
  <h1>
    Hello, world!
  </h1>
  <h2>
    It is 
    8:05:00 AM
    .
  </h2>
</div>
`;

exports[`test use act 2`] = `
<div>
  <h1>
    Hello, world!
  </h1>
  <h2>
    It is 
    8:05:05 AM
    .
  </h2>
</div>
`;
